agent/api/task/service_connect.go

```diff
@@ -65,3 +81,246 @@ type DNSConfigEntry struct {
+		if err := validateCIDR(trimmedIpv6cidr, "IPV6CIDR"); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// validateIngressConfig validates the service connect ingress config based on network mode.
+func validateIngressConfig(scIngressConfigList []IngressConfigEntry, taskNetworkMode string) error {
+	// IngressConfig can be empty since an ECS service can only act as a client
+	if len(scIngressConfigList) == 0 {
+		return nil
+	}
+	// AWSVPC network mode
+	switch taskNetworkMode {
+	case BridgeNetworkMode:
+		if err := validateBridgeIngressConfig(scIngressConfigList); err != nil {
+			return err
+		}
+	case AWSVPCNetworkMode:
+		if err := validateAwsVpcIngressConfig(scIngressConfigList); err != nil {
+			return err
+		}
+	default:
+		return fmt.Errorf("service connect config validation: service connect does not support %s newtork mode", taskNetworkMode)
+	}
+	return nil
+}
+
+// validateAwsVpcIngressConfig validates the service connect ingress config with awsvpc network mode.
+func validateAwsVpcIngressConfig(scIngressConfigList []IngressConfigEntry) error {
+	portsMap := map[uint16]bool{}
+	interceptPortValue := uint16(0)
+	listenerPortValue := uint16(0)
+	for _, entry := range scIngressConfigList {
+		// verify host port does not exist in awsvpc mode
+		if entry.HostPort != nil {
+			return fmt.Errorf("service connect config validation: host port should not exist for awsvpc mode")
+		}
+		// verify intercept ports have no conflicts
+		if entry.InterceptPort != nil {
+			interceptPortValue = aws.Uint16Value(entry.InterceptPort)
+			if err := validatePort(interceptPortValue, AWSVPCNetworkMode); err != nil {
+				return err
+			}
+			if entry.ListenerName == "" {
+				return fmt.Errorf("service connect config validation: no listener name")
+			}
+			if present := portsMap[interceptPortValue]; present {
+				return fmt.Errorf("service connect config validation: intercept port %d conflicts", interceptPortValue)
+			}
+			portsMap[interceptPortValue] = true
+		}
+		// verify listener ports have no conflicts
+		if entry.ListenerPort >= uint16(1) {
+			listenerPortValue = entry.ListenerPort
+			if present := portsMap[listenerPortValue]; present {
+				return fmt.Errorf("service connect config validation: listener port %d conflicts", listenerPortValue)
+			}
+			portsMap[listenerPortValue] = true
+		}
+	}
+	return nil
+}
+
+// validateBridgeIngressConfig validates the service connect ingress config with bridge network mode.
+func validateBridgeIngressConfig(scIngressConfigList []IngressConfigEntry) error {
+	hostPortsMap := map[uint16]bool{}
+	hostPortValue := uint16(0)
+	listenerPortsMap := map[uint16]bool{}
+	listenerPortValue := uint16(0)
+	for _, entry := range scIngressConfigList {
+		// verify intercept port does not exist in bridge mode
+		if entry.InterceptPort != nil {
+			return fmt.Errorf("service connect config validation: intercept port should not exist for bridge mode")
```

Comment:
same here, I feel this should be a warning. Although not positive. Would like others to chime in.

Reply:
Yes. +1 to warning and skip